src/backend/XmlHttpRequest.ts

```diff
@@ -69,11 +75,12 @@ export default class XmlHttpRequest extends BaseFileSystem implements FileSystem
     if (typeof(opts.index) === "string") {
       XmlHttpRequest.FromURL(opts.index, cb, opts.baseUrl, false);
     } else {
-      cb(null, new XmlHttpRequest(opts.index, opts.baseUrl, false));
+      cb(null, new XmlHttpRequest(opts.index, opts.baseUrl, false, opts.use));
     }
   }
   public static isAvailable(): boolean {
-    return typeof(XMLHttpRequest) !== "undefined" && XMLHttpRequest !== null;
+    return (typeof(fetch) !== "undefined" && fetch !== null) ||
+           (typeof(XMLHttpRequest) !== "undefined" && XMLHttpRequest !== null);
```

Comment:
Note: `supportsSynch()` needs to be updated in the Node case. If `XMLHttpRequest` isn't available, there is no way to synchronously download files, and `supportsSynch()` should return `false`.